apps/req.c

```diff
@@ -545,6 +545,15 @@ int req_main(int argc, char **argv)
                     goto end;
                 }
             }
+
+#ifndef OPENSSL_NO_EC
+            if (pkey_type == EVP_PKEY_EC
+                    && EVP_PKEY_CTX_get_ec_keygen_bits(genctx, &newkey) <= 0) {
```

Comment:
I think I would prefer that simpler approach. The other advantage is that it could be more easily backported to 1.0.2. Perhaps in some future version we could add this new API.